src/components/permissions/rowLevelPolicyHelpers.ts

```diff
@@ -0,0 +1,332 @@
+    policiesObject = {
+      ...policiesObject,
+      [permissionNameAbbreviation]: compileRowLevelPolicies(
+        permissionNameAbbreviation,
+        permissionPolicyRules
+      ),
+    }
+    // It's possible to have permission policy linked to permission name without any templatePermissionIds
+    if (!templatePermissionId) return
+
+    const templatePermissionAbbreviation = getTemplatePermissionAbbreviation(permissionRow)
+    policiesObject = {
+      ...policiesObject,
+      [templatePermissionAbbreviation]: compileRowLevelPolicies(
+        templatePermissionAbbreviation,
+        permissionPolicyRules
+      ),
+    }
+  })
+  // Turn from object (of keys and arrays) to flattened array
+  const policies = Object.values(policiesObject).flat()
+
+  return policies
+}
+
+/* Generates row level policies, for one set of rules (for one PermissionRow)
+  in "pp2pn2tp2", {
+      application: {
+        view: {
+          template_id: 'jwtPermission_bigint_templateId',
+          user_id: 'jwtUserDetails_bigint_userId',
+        },
+      },
+    }
+  out [
+    CREATE POLICY "view_pp2pn2tp2" ON "application" FOR SELECT USING (jwt_get_boolean('pp2pn2tp2') = true and user_id = jwt_get_bigint('userId') AND template_id = jwt_get_bigint('pp2pn2tp2_templateId'))
+  ]
+*/
+const compileRowLevelPolicies = (permissionAbbreviation: string, permissionPolicyRules: object) => {
+  const policies: Array<string> = []
+
+  Object.entries(permissionPolicyRules).forEach(([tableName, rulesByType]) => {
+    let usingCondition = 'true'
+    const { view: viewRules } = rulesByType
+    // Need to figure out USING clause for UPDATE rule that defines WITH CHECK via rules (for UPDATE, see compileRowLevelPolicy for more details)
+    // So the UPDATE policy will use USING clause from view (select) rules
+    if (viewRules)
+      usingCondition = replacePlaceholders(
+        getSqlConditionFromJSON(viewRules),
+        permissionAbbreviation
+      )
+
+    Object.entries(rulesByType).forEach(([ruleType, rules]) => {
+      // replacePlaceholders would replace `jwtUserDetails_bigint_${something}` with `jwt_get_bigint_${something}` etc..
+      const condition = replacePlaceholders(getSqlConditionFromJSON(rules), permissionAbbreviation)
+
+      policies.push(
+        compileRowLevelPolicy(
+          tableName,
+          permissionAbbreviation,
+          ruleType,
+          usingCondition,
+          condition
+        )
+      )
+    })
+  })
+
+  return policies
+}
+
+// Constants for compileRowLevelPolicy
+// https://www.postgresql.org/docs/current/sql-createpolicy.html
+const ruleTypes: RuleTypes = {
+  view: {
+    for: 'SELECT',
+    using: true,
+    withCheck: false,
+  },
+  update: {
+    for: 'UPDATE',
+    using: true,
+    withCheck: true,
+  },
+  create: {
+    for: 'CREATE',
+    using: false,
+    withCheck: true,
+  },
+  delete: {
+    for: 'DELETE',
+    using: true,
+    withCheck: false,
+  },
+}
```

Comment:
In separated file file would be nice

Reply:
Can move it to that utility methods file. I thought to keep it right above where it's used, so it's easer to see what `const ruleSettings = ruleTypes[ruleType]` is doing. It's quite difficult to try and find a readability balance (between abstracting logic into methods/variables and using those methods/variables in a place where you can't directly see what those methods/variables do)

Reply:
60db69257ccc33df3bd930a57d334fe42862b8bf

Reply:
Ok, changed it further as per telegram chat:
4ca8a2d7ce673dfc9e7fb0f219fe9d6fcf69e2d8

Reply:
Btw, from my understanding `model` is kind of like the schema, the description of shape of DB. In this case we have a constant. Same same in terms of your comment but dont' want to get confused with the terms